src/lib/builders/table-of-contents/create.ts

```diff
@@ -0,0 +1,259 @@
+                visibleElementIdxs.set(tempVisibleElementIdxs);
+
+                // Remove all parents of obsIndex from the activeParentIdxs list.
+                if ((tocType === 'highest-parents' || tocType === 'lowest-parents') && headingParentsLU[toc_idx]) {
+                    headingParentsLU[toc_idx]?.forEach((parent: number) => {
+                        const tempActiveParentIdxs = get(activeParentIdxs);
+                        const index = tempActiveParentIdxs.indexOf(parent);
+                        tempActiveParentIdxs.splice(index, 1);
+                        activeParentIdxs.set(tempActiveParentIdxs);
+                    });
+                }
+            }
+        }
+
+        const allActiveHeaderIdxs = Array.from(new Set(get(visibleElementIdxs).map((idx) => elementHeadingLU[idx])));
+
+        let activeHeaderIdxs: number[];
+
+        if (allActiveHeaderIdxs.length === 0) {
+            activeHeaderIdxs = [];
+        } else if (tocType === 'highest') {
+            activeHeaderIdxs = [Math.min(...allActiveHeaderIdxs)];
+        } else if (tocType === 'lowest') {
+            activeHeaderIdxs = [Math.max(...allActiveHeaderIdxs)];
+        } else if (tocType === 'all-active') {
+            activeHeaderIdxs = allActiveHeaderIdxs;
+        } else {
+            const activeHeaderIdx = tocType === 'highest-parents' ? Math.min(...allActiveHeaderIdxs) : Math.max(...allActiveHeaderIdxs);
+
+            if (headingParentsLU[activeHeaderIdx]) {
+                activeHeaderIdxs = [...<[]>headingParentsLU[activeHeaderIdx], activeHeaderIdx];
+            } else {
+                activeHeaderIdxs = [activeHeaderIdx];
+            }
+        }
+
+        // Set store to active indexes.
+        activeHeadingIdxs.set(activeHeaderIdxs);
+    }
+
+    const item = builder('', {
+        returned: () => {
+            return {
+                role: 'link',
+                tabindex: 0
+            }
+        },
+        action: (node: HTMLElement, params: NavItemProps) => {
+            const { id } = params;
+
+            const unsub = executeCallbacks(
+                addEventListener(node, 'click', (e) => {
+                    e.preventDefault();
+                    scrollToElement(`#${id}`);
+                }),
+            );
+
+            return {
+                destroy: unsub
+            };
+        }
+    });
+
+    onMount(() => {
+        // Init
+        generateInitialLists();
+        findParentIdxs();
+        createElementHeadingLU();
+
+        headingsTree.set(createTree(headingsList));
+
+        // Create observer and observe all elements.
+        observer = new IntersectionObserver(handleElementObservation, { root: null, threshold: observer_threshold });
+        elementsList.forEach((el) => observer?.observe(el));
+
+        return () => {
+            observer?.disconnect();
+        };
+    });
+
+    return {
+        activeHeadingIdxs,
+        headingsTree,
+        navItem: item
```

Comment:
I think it can be just item tbh

Reply:
I agree. Then I'll rename the `items` in the `TableOfContentsItem` type to `children` just so we don't have the same name twice